jsonschema/src/keywords/pattern.rs

```diff
@@ -43,21 +41,25 @@ impl Validate for PatternValidator {
         instance_path: &InstancePath,
     ) -> ErrorIterator<'a> {
         if let Value::String(item) = instance {
-            if !self.pattern.is_match(item) {
-                return error(ValidationError::pattern(
-                    instance_path.into(),
-                    instance,
-                    self.original.clone(),
-                ));
+            if let Ok(is_match) = self.pattern.is_match(item) {
```

Comment:
If `Err` is returned, we should indicate this backtracking problem to the user with a new `ValidationError` variant.

Reply:
Yes, good idea. I'll add this.